### PR TITLE
Add F3D_PLUGINS_PATH

### PR DESCRIPTION
**Pull request by Meakk**

### CODE REVIEW

File: library/src/engine.cxx [hunk 1]
```diff
@@ -143,32 +143,62 @@ void engine::loadPlugin(const std::string& pathOrName)
...
+      std::string envValue;
+      if (vtksys::SystemTools::GetEnv("F3D_PLUGINS_PATH", envValue))
+      {
+        // split path with OS separator (':' on Linux/macOS and ';' on Windows)
+#ifdef _WIN32
+        char sep = ';';
+#else
+        char sep = ':';
+#endif
+        vtksys::SystemTools::Split(envValue, searchPaths, sep);
```
Comment (mwestphal) [inline library/src/engine.cxx:157]: use filesystem instead ?

Comment (Meakk) [reply library/src/engine.cxx:157]: It's not related to fs here, it's just a split of a string to a string list.

**Comment by mwestphal** [comment]
This is maybe testable?

File: doc/libf3d/PLUGINS.md [hunk 2]
```diff
@@ -58,4 +58,9 @@ The list of existing mimetypes can be find [here](https://www.iana.org/assignmen
 
 ## Loading your plugin
 
-The plugin can be loaded using `f3d::engine::loadPlugin("path or name")` api if you are using libf3d, or `--load-plugins="path or name"` option if you are using F3D application.
+The plugin can be loaded using `f3d::engine::loadPlugin("path or name")` api if you are using libf3d, or `--load-plugins="path or name"` option if you are using F3D application. Several plugins can be specified by giving a comma-separated list.
```
Comment (mwestphal) [inline doc/libf3d/PLUGINS.md:61]: this needs to be documented in user doc, not (only) libf3d doc.

File: doc/libf3d/PLUGINS.md [hunk 3]
```diff
@@ -58,4 +58,9 @@ The list of existing mimetypes can be find [here](https://www.iana.org/assignmen
 
 ## Loading your plugin
 
-The plugin can be loaded using `f3d::engine::loadPlugin("path or name")` api if you are using libf3d, or `--load-plugins="path or name"` option if you are using F3D application.
+The plugin can be loaded using `f3d::engine::loadPlugin("path or name")` api if you are using libf3d, or `--load-plugins="path or name"` option if you are using F3D application. Several plugins can be specified by giving a comma-separated list.  
+Here is how the plugins are searched (by precedence order):  
+1. Consider the option given is a full path
+2. Search in the paths specified in `F3D_PLUGINS_PATH` environment variable
+3. Search in the same directory than `f3d.exe` (Windows only)
+4. Rely on OS specific paths (e.g. `LD_LIBRARY_PATH` on Linux)
```
Comment (mwestphal) [inline doc/libf3d/PLUGINS.md:66]: what about macOS ?

File: library/src/engine.cxx [hunk 4]
```diff
@@ -143,32 +143,62 @@ void engine::loadPlugin(const std::string& pathOrName)
...
+          break;
+        }
+      }
+
       if (!handle)
       {
-        throw engine::plugin_exception(
-          "Cannot open the library \"" + pathOrName + "\": " + vtksys::DynamicLoader::LastError());
+        // Rely on internal system (e.g. LD_LIBRARY_PATH on Linux) by giving only the file name
+        handle = vtksys::DynamicLoader::OpenLibrary(libName);
```
Comment (mwestphal) [inline library/src/engine.cxx:196]: maybe give more details about what DynamicLoader do ? its doc is hard to find.

File: library/src/engine.cxx [hunk 5]
```diff
@@ -143,32 +143,62 @@ void engine::loadPlugin(const std::string& pathOrName)
...
+      {
+        std::string tryPath = path + '/' + libName;
+        tryPath = vtksys::SystemTools::ConvertToOutputPath(tryPath);
+
+        handle = vtksys::DynamicLoader::OpenLibrary(tryPath);
+
+        if (handle)
+        {
+          // plugin is found and loaded
+          break;
```
Comment (mwestphal) [inline library/src/engine.cxx:189]: could we add an example test relying on this mechanism ?

File: doc/libf3d/PLUGINS.md (hunk 2 above)
Comment (Meakk) [reply doc/libf3d/PLUGINS.md:61]: I'll move the precedence order in the user documentation, it makes more sense.

File: library/src/engine.cxx (hunk 4 above)
Comment (Meakk) [reply library/src/engine.cxx:196]: it's just a `dlopen`, nothing fancy

File: library/src/engine.cxx (hunk 5 above)
Comment (Meakk) [reply library/src/engine.cxx:189]: I don't see any easy way to cover this.

**Review by mwestphal — APPROVED** [review]

**Pull request merged**